addons/mail/static/tests/owl/component/chatter_tests.js

```diff
@@ -219,6 +263,82 @@ QUnit.test('show attachment box', async function (assert) {
     );
 });
 
+QUnit.test('composer show/hide on log note/send message', async function (assert) {
+    assert.expect(10);
+
+    await this.start();
+    const chatterLocalId = this.env.store.dispatch('createChatter', {
+        initialThreadId: 100,
+        initialThreadModel: 'res.partner',
+    });
+    await this.createChatterComponent({ chatterLocalId });
+    assert.strictEqual(
+        document.querySelectorAll(`.o_Chatter`).length,
+        1,
+        "should have a chatter"
+    );
+    assert.strictEqual(
+        document.querySelectorAll(`.o_ChatterTopbar`).length,
+        1,
+        "should have a chatter topbar"
+    );
```

Comment:
Kinda redundant with previous test.

Qunit runs test from top to bottom (except on re-running with failed tests, in that case failed tests are run first), so I would say if a former test make an assertion, ideally future tests shouldn't make the same...

...But at the same time, if we decide to remove/update some tests in the future, it may become harder to understand why some tests fail, as they made assumptions on previous valid assertions from other tests...

IMO, if you doubt of this scenario, I'm totally okay to keep lots of "dummy" assertions in the beginning!

Reply:
I understand your point.
IMO when you read a test you don't always read the previous ones so I think the test should test all the things he needs to use. In that example, I need to be sure `buttonSendMessage` and `buttonLogNote` exist so I made asserts top-down to check their existence. 

By the way, I agree that following assert will fail if there is no chatter or no chatter topbar so the first two asserts can be removed.

```
assert.strictEqual(
    document.querySelectorAll(`.o_ChatterTopbar_buttonSendMessage`).length,
    1,
    "should have a send message button"
);
```